flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -238,6 +242,16 @@ public void failJobIfNotEnoughResources() throws Exception {
 		waitForTermination(scheduler);
 		final JobStatus jobStatus = scheduler.requestJobStatus();
 		assertThat(jobStatus, is(Matchers.equalTo(JobStatus.FAILED)));
+
+		Throwable failureCause = scheduler.requestJob()
+			.getFailureInfo()
+			.getException()
+			.deserializeError(DefaultSchedulerTest.class.getClassLoader());
+		assertTrue(findThrowable(failureCause, NoResourceAvailableException.class).isPresent());
+		assertTrue(
+			findThrowableWithMessage(
+				failureCause,
+				"Could not allocate the required slot within slot request timeout.").isPresent());
```

Comment:
The error message does not mention the number of slots anymore. IIRC in the legacy code path, the error message does not consider slot sharing so the information provided to the user was not accurate. I think it's fine to change this behavior. WDYT, @tillrohrmann?

Reply:
It is true that we always report the number of required logical slots which can be an order larger than the underlying physical slots due to slot sharing. I guess that we re-add the proper reporting about the number of required slots available slots once we schedule based on pipelined regions. Until then, I'm ok with changing the message here.